packages/frontend-common/formats/chart/hierarchical-tree/HierarchicalTreeView.tsx

```diff
@@ -0,0 +1,143 @@
+        data: formatData,
+    });
+
+    const getNodeColor = useCallback(() => {
+        return colors || '#000000';
+    }, [colors]);
+
+    return (
+        <Box sx={{ height: `500px`, position: 'relative' }} role="tree">
+            <FormatFullScreenMode>
+                <Box ref={parentRef} sx={{ width: '100%', height: '100%' }}>
+                    <Suspense>
+                        <TreeView
+                            ref={treeRef}
+                            data={tree}
+                            orientation={orientation}
+                            translate={treeTranslate}
+                            renderCustomNodeElement={(props) => (
+                                <HierarchicalTreeNode
+                                    {...props}
+                                    orientation={orientation}
+                                    width={nodeWidth ?? DEFAULT_NODE_WIDTH}
+                                    height={nodeHeight ?? DEFAULT_NODE_HEIGHT}
+                                    getNodeColor={getNodeColor}
+                                />
+                            )}
+                            nodeSize={{
+                                x: (nodeWidth ?? DEFAULT_NODE_WIDTH) + 32,
+                                y: (nodeHeight ?? DEFAULT_NODE_HEIGHT) + 32,
+                            }}
+                            scaleExtent={{
+                                min:
+                                    minimumScaleValue ??
+                                    DEFAULT_MINIMUM_SCALE_VALUE,
+                                max:
+                                    maximumScaleValue ??
+                                    DEFAULT_MAXIMUM_SCALE_VALUE,
+                            }}
+                            zoom={initialZoom}
+                            initialDepth={initialDepth}
+                            hasInteractiveNodes
+                            onNodeClick={centerOnNode}
+                            centeringTransitionDuration={150}
+                            dimensions={dimensions}
+                        />
+                    </Suspense>
+                </Box>
+
+                <div
+                    style={{
+                        position: 'absolute',
+                        bottom: 0,
+                        left: 0,
+                    }}
+                >
+                    {<MouseIcon />}
+                </div>
+            </FormatFullScreenMode>
+        </Box>
+    );
+}
+
+type HierarchicalTreeViewProps = {
+    formatData: Datum[];
+
+    orientation?: 'horizontal' | 'vertical';
+    nodeWidth?: number | null;
+    nodeHeight?: number | null;
+    spaceBetweenNodes?: number;
+
+    initialZoom?: number;
+    initialDepth?: number;
+
+    minimumScaleValue?: number | null;
+    maximumScaleValue?: number | null;
+    colors?: string;
+};
+
+// @ts-expect-error TS2345
+export default compose(injectData())(HierarchicalTreeView);
```

Comment:
Can you type it by passing the props type ads generic
Something like:

```suggestion
export default compose<HierarchicalTreeViewProps, Omit<HierarchicalTreeViewProps, 'formatData' | ...>>(injectData())(HierarchicalTreeView);
```